util/fipstools/inject_hash/macho_parser/macho_parser.c

```diff
@@ -0,0 +1,165 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+#include <stdint.h>
+
+#include "common.h"
+#include "macho_parser.h"
+
+// Documentation for the Mach-O structs can be found in macho-o/loader.h and mach-o/nlist.h
+int read_macho_file(const char *filename, machofile *macho) {
+    FILE *file = NULL;
+    load_cmd *load_commands = NULL;
+    int ret = 1;
```

Comment:
From a principal point of view, I prefer to start with assuming an error state. Makes it easier to handle errors without accidentally forgetting to switch to an error state before returning.